### PR TITLE
refactor(settings-row): convert files to typescript

### PR DESCRIPTION
**Pull request by tomdavies73**
### Proposed behaviour

All component files should be refactored and converted to TypeScript.

### Current behaviour

Component files are currently JavaScript.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/settings-row/settings-row-test.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  title,
+  headingType,
+  children,
+  description,
+  divider,
+  className,
+  ...args
+}: Partial<SettingsRowProps>) => {
+  return (
+    <>
```
Comment (DipperTheDan) [inline src/components/settings-row/settings-row-test.stories.tsx:26]: I don't think this fragment is needed here.

File: src/components/settings-row/settings-row.spec.tsx [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
+import Heading, { HeadingType } from "../heading";
 import { rootTagTest } from "../../__internal__/utils/helpers/tags/tags-specs";
 import {
   assertStyleMatch,
   testStyledSystemMargin,
 } from "../../__spec_helper__/test-utils";
 
 describe("SettingsRow", () => {
-  testStyledSystemMargin((props) => <SettingsRow {...props} />, { m: "0" });
+  testStyledSystemMargin((props: any) => <SettingsRow {...props} />, {
```
Comment (DipperTheDan) [inline src/components/settings-row/settings-row.spec.tsx:16]:
```suggestion
  testStyledSystemMargin((props) => <SettingsRow {...props} />, {
```

File: src/components/settings-row/settings-row.spec.tsx [hunk 3]
```diff
@@ -1,29 +1,31 @@
...
+  });
 
   describe("render", () => {
     const name = "foobar-row";
     const title = "Some Title";
     const description = <span>Some descriptive text</span>;
     const childId = "my_child";
     const children = <span id={childId} />;
-    let wrapper = shallow(
+    let wrapper: any;
```
Comment (DipperTheDan) [inline src/components/settings-row/settings-row.spec.tsx:26]:
```suggestion
    let wrapper:  ReactWrapper | ShallowWrapper;
```

File: src/components/settings-row/settings-row.spec.tsx [hunk 4]
```diff
@@ -59,7 +61,7 @@ describe("SettingsRow", () => {
     });
 
     describe("when description is provided", () => {
-      let head;
+      let head: any;
```
Comment (DipperTheDan) [inline src/components/settings-row/settings-row.spec.tsx:64]:
```suggestion
      let head: ShallowWrapper;
```

File: src/components/settings-row/settings-row.spec.tsx [hunk 5]
```diff
@@ -1,29 +1,31 @@
 import React from "react";
 import { shallow, mount } from "enzyme";
```
Comment (DipperTheDan) [inline src/components/settings-row/settings-row.spec.tsx:2]:
```suggestion
import { shallow, mount, ReactWrapper, ShallowWrapper } from "enzyme";
```

File: src/components/settings-row/settings-row-test.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+export const Default = ({
+  title,
+  headingType,
+  children,
+  description,
+  divider,
+  className,
+  ...args
+}: Partial<SettingsRowProps>) => {
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/settings-row/settings-row-test.stories.tsx:24]: Not sure you need `...args` here either

**Review by DipperTheDan — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**